### PR TITLE
[WIP] Circleci updates / changes

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by chriseth** [comment]
Hm, build times are fast enough for us I think. What we are missing is a recent eth binary that does not depend on libmicrohttpd anymore so we can run the end to end tests.

**Comment by axic** [comment]
@chfast this was your old branch on the repo. Can you please have a look to see if any of these changes still make sense?

File: cmake/EthCompilerSettings.cmake [hunk 1]
```diff
@@ -186,11 +186,8 @@ else ()
 	message(WARNING "Your compiler is not tested, if you run into any issues, we'd welcome any patches.")
 endif ()
 
-if (SANITIZE)
+if(SANITIZE)
```
Comment (chfast) [inline cmake/EthCompilerSettings.cmake:189]: This is a clean up if you don't have the sanitizer-blacklist.txt file.

File: scripts/install_deps.sh [hunk 2]
```diff
@@ -165,11 +165,12 @@ case $(uname -s) in
...
 #------------------------------------------------------------------------------
 
-            Debian)
+            Debian*)
                 #Debian
+                . /etc/os-release
                 install_z3=""
-                case $(lsb_release -cs) in
-                    wheezy)
+                case $VERSION_ID in
```
Comment (chfast) [inline scripts/install_deps.sh:172]: This is a way of doing the same if you don't have `lsb_release` tool (common in docker images).

**Review by chfast — APPROVED** [review]

**Review by chriseth — CHANGES_REQUESTED** [review]
This has to be rebased, the diff is really weird.

**Comment by axic** [comment]
Pulled out the lsb change (#3612). The two remaining useful changes from here are:
- sanitizer change
- using ninja in cireclci

@chfast @chriseth any of these are worth merging?

**Comment by chfast** [comment]
I don't think so. The sanitizer changes are in https://github.com/ethereum/cable, so you may consider integrating this when stable enough.

Ninja speeds up builds, so you can enabled it in CIs. But you don't need this PR.

**Comment by chriseth** [comment]
I think our build is already fast enough on circle. We need to speed up the testing :)

**Comment by chfast** [comment]
I'm not pushing Ninja.
BTW, Ninja also has higher requirements for correct describe dependencies, so you might find some issues in CMake scripts.

**Pull request closed without merging**